docs/advanced-usage/enabling-natural-language-search.md

```diff
@@ -0,0 +1,129 @@
+---
+icon: kolena/comparison-16
+---
+
+# :kolena-comparison-20: Enabling Natural Language Search
+
+Kolena supports natural language and similar image search across [`Image`][kolena.workflow.Image] data previously registered to the platform.
+Users may enable this functionality by extracting and uploading the corresponding search embeddings using a Kolena provided package.
+
+!!! note end
+    Kolena supports search embedding extraction and upload as an opt-in feature for our customers.
+    Please message your point of contact for the latest relevant extractor package.
+
+## Example
+
+The `kolena/kolenaIO` contains a runnable [example](https://github.com/kolenaIO/kolena/tree/trunk/examples/search_embeddings)
+of integration code for embeddings extraction and upload.
+This builds off the data uploaded in the [age_estimation](https://github.com/kolenaIO/kolena/tree/trunk/examples/age_estimation)
+example workflow, and is best run after this data has been uploaded to your Kolena platform.
+
+## How to Enabling Natural Language and Search on Kolena
+
+Uploading embeddings to Kolena can be done in three simple steps:
+
+- [**Step 1**](#step-1-install-kolena_embeddings-package): installing dependency package
+- [**Step 2**](#step-2-load-images-for-extraction): loading images for input to extraction library
+- [**Step 3**](#step-3-extract-and-upload-embeddings): extracting and uploading search embeddings
+
+Let's take a look at each step with example code snippets.
+
+### Step 1: Install `kolena_embeddings` Package
+
+Copy the `kolena_embeddings-*.*.*.tar.gz` file (provided by your Kolena contact) to your working directory, and install it as a dependency.
+
+=== "`pip`"
+
+    ```shell
+    pip install ./kolena_embeddings-*.*.*.tar.gz
+    ```
+
+=== "`poetry`"
+
+    ```shell
+    poetry add ./kolena_embeddings-*.*.*.tar.gz
+    ```
+
+This package provides the `kclip.util.extract_and_upload_embeddings` method:
```

Comment:
Just realized we still have this as `kclip` -- thoughts on if this is an issue?

Reply:
Renamed to kembeddings -- also putting together associated PR from the corresponding `kolena embeddings` repo